source/parse.h

```diff
@@ -6909,7 +6920,41 @@ class parser
             next();
         }
 
-        if (auto id = qualified_id()) {
+        if (auto& c = curr();
+            c == "decltype"
+            )
+        {
+            if (
+                peek(1) && peek(1)->type() == lexeme::LeftParen
+                && peek(2) && *peek(2) == "auto"
+                && peek(3) && peek(3)->type() == lexeme::RightParen)
+            {
+                error(
+                    "decltype(auto) is not needed in Cpp2 - use 'forward: _' parameters/returns/objects instead"
+                    "\n  (temporary alpha limitation) returns means auto&&"
+                    "\n  (temporary alpha limitation) objects are not yet supported",
```

Comment:
Besides any conflict resolution,
I think you'll just need to update this error message.

Reply:
Done, thanks!